Tests fail using BeautifulSoup4 v4.12.2

Hi, \
I recently upgraded bs4 from 4.11.2 to 4.12.2 and now pycaption fails during testing. \
I'm not exactly sure where the issue is, so I'm starting here and maybe you can reproduce it or give some insight into the kind of error.

Can you verify the issue?

#### Additional information
Going through the [bs4 changelog](https://git.launchpad.net/beautifulsoup/tree/CHANGELOG), I didn't notice any change related to this issue. \
I have also rebuilt pycaption with bs4 4.11.2 and the tests ran through just fine. My distro doesn't package 4.12.0 or 4.12.1, but if necessary, I might do some bisecting.

#### System information
OS: Arch Linux
Kernel: 6.3.2-arch1-1
Python: 3.11.3-1
Pycaption: 2.1.0/2.1.1 ([AUR package](https://aur.archlinux.org/packages/python-pycaption) / manually updated to 2.1.1)
BeautifulSoup4: 4.11.2-3/4.12.2-1 ([Official package](https://archlinux.org/packages/extra/any/python-beautifulsoup4/))

#### Appendix 1 (full error log)
```
==> Starting check()...
============================= test session starts ==============================
platform linux -- Python 3.11.3, pytest-7.3.1, pluggy-1.0.0
rootdir: /build/python-pycaption/src/pycaption-2.1.0
plugins: lazy-fixture-0.6.3
collected 256 items

tests/test_base.py .......                                               [  2%]
tests/test_dfxp.py .......................................               [ 17%]
tests/test_dfxp_conversion.py F

=================================== FAILURES ===================================
________________________ TestDFXPtoDFXP.test_conversion ________________________

self = <tests.test_dfxp_conversion.TestDFXPtoDFXP object at 0x7fbfff47abd0>
sample_dfxp_output = '<?xml version="1.0" encoding="utf-8"?>\n<tt xml:lang="en" xmlns="http://www.w3.org/ns/ttml" xmlns:tts="http://www.w3....00:00:36.200" region="bottom" style="default">\n    &lt;LAUGHING &amp; WHOOPS!&gt;\n   </p>\n  </div>\n </body>\n</tt>'
sample_dfxp = '<?xml version="1.0" encoding="utf-8"?>\n<tt xml:lang="en" xmlns="http://www.w3.org/ns/ttml"\n    xmlns:tts="http://ww...00:00:36.200" region="bottom" style="default">\n    &lt;LAUGHING &amp; WHOOPS!&gt;\n   </p>\n  </div>\n </body>\n</tt>'

    def test_conversion(self, sample_dfxp_output, sample_dfxp):
        caption_set = DFXPReader().read(sample_dfxp)
        results = DFXPWriter().write(caption_set)
    
        assert isinstance(results, str)
>       self.assert_dfxp_equals(sample_dfxp_output, results)

tests/test_dfxp_conversion.py:25: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tests.test_dfxp_conversion.TestDFXPtoDFXP object at 0x7fbfff47abd0>
first = '<?xml version="1.0" encoding="utf-8"?>\n<tt xml:lang="en" xmlns="http://www.w3.org/ns/ttml" xmlns:tts="http://www.w3....00:00:36.200" region="bottom" style="default">\n    &lt;LAUGHING &amp; WHOOPS!&gt;\n   </p>\n  </div>\n </body>\n</tt>'
second = '<?xml version="1.0" encoding="utf-8"?>\n<tt xml:lang="en" xmlns="http://www.w3.org/ns/ttml" xmlns:tts="http://www.w3....:00:36.200" region="bottom" style="default">\n    &lt;LAUGHING &amp; WHOOPS!&gt;\n   </p>\n  </div>\n </body>\n</tt>\n'
ignore_styling = False, ignore_spans = False

    def assert_dfxp_equals(self, first, second,
                           ignore_styling=False,
                           ignore_spans=False):
        first_soup = BeautifulSoup(first, 'lxml')
        second_soup = BeautifulSoup(second, 'lxml')
    
        if ignore_styling:
            self._remove_styling(first_soup)
            self._remove_styling(second_soup)
    
        if ignore_spans:
            self._remove_spans(first_soup)
            self._remove_spans(second_soup)
    
        self._trim_text(first_soup)
        self._trim_text(second_soup)
    
>       assert first_soup == second_soup
E       AssertionError

tests/mixins.py:140: AssertionError
=========================== short test summary info ============================
FAILED tests/test_dfxp_conversion.py::TestDFXPtoDFXP::test_conversion - Asser...
!!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!
================== 1 failed, 46 passed, 18 warnings in 0.86s ===================
==> ERROR: A failure occurred in check().

```

--- a/setup.py
+++ b/setup.py
@@ -8,7 +8,7 @@ README_PATH = os.path.join(
 )
 
 dependencies = [
-    'beautifulsoup4>=4.8.1',
+    'beautifulsoup4>=4.12.1',
     'lxml>=4.9.1',
     'cssutils>=2.0.0',
 ]

--- a/tests/fixtures/dfxp.py
+++ b/tests/fixtures/dfxp.py
@@ -47,7 +47,8 @@ def sample_dfxp():
    </p>
   </div>
  </body>
-</tt>"""
+</tt>
+"""
 
 
 @pytest.fixture(scope="session")
@@ -260,7 +261,8 @@ def sample_dfxp_with_relativized_positioning():
    </p>
   </div>
  </body>
-</tt>"""
+</tt>
+"""
 
 
 @pytest.fixture(scope="session")
@@ -348,7 +350,8 @@ def sample_dfxp_from_sami_with_positioning():
    </p>
   </div>
  </body>
-</tt>"""
+</tt>
+"""
 
 
 @pytest.fixture(scope="session")
@@ -408,7 +411,8 @@ def sample_dfxp_long_cue_fit_to_screen():
    </p>
   </div>
  </body>
-</tt>"""
+</tt>
+"""
 
 
 @pytest.fixture(scope="session")
@@ -431,7 +435,8 @@ def sample_dfxp_from_sami_with_margins():
    </p>
   </div>
  </body>
-</tt>"""
+</tt>
+"""
 
 
 @pytest.fixture(scope="session")
@@ -454,7 +459,8 @@ def sample_dfxp_from_sami_with_lang_margins():
    </p>
   </div>
  </body>
-</tt>"""
+</tt>
+"""
 
 
 @pytest.fixture(scope="session")
@@ -476,7 +482,8 @@ def sample_dfxp_from_sami_with_span():
    </p>
   </div>
  </body>
-</tt>"""
+</tt>
+"""
 
 
 @pytest.fixture(scope="session")
@@ -499,7 +506,8 @@ def sample_dfxp_from_sami_with_bad_span_align():
    </p>
   </div>
  </body>
-</tt>"""
+</tt>
+"""
 
 
 @pytest.fixture(scope="session")
@@ -597,7 +605,8 @@ def sample_dfxp_invalid_but_supported_positioning_output():
    </p>
   </div>
  </body>
-</tt>"""
+</tt>
+"""
 
 
 @pytest.fixture(scope="session")
@@ -685,7 +694,8 @@ def sample_dfxp_multiple_regions_output():
    </p>
   </div>
  </body>
-</tt>"""
+</tt>
+"""
 
 
 @pytest.fixture(scope="session")
@@ -826,7 +836,8 @@ def sample_dfxp_output():
    </p>
   </div>
  </body>
-</tt>"""
+</tt>
+"""
 
 
 @pytest.fixture(scope="session")
@@ -886,7 +897,8 @@ def sample_dfxp_style_tag_with_no_xml_id_output():
    </p>
   </div>
  </body>
-</tt>"""
+</tt>
+"""
 
 
 @pytest.fixture(scope="session")
@@ -960,7 +972,8 @@ def sample_dfxp_from_scc_output():
    </p>
   </div>
  </body>
-</tt>"""
+</tt>
+"""
 
 
 @pytest.fixture(scope="session")
@@ -1009,7 +1022,8 @@ def sample_dfxp_with_properly_closing_spans_output():
    </p>
   </div>
  </body>
-</tt>"""
+</tt>
+"""
 
 
 @pytest.fixture(scope="session")
@@ -1109,7 +1123,8 @@ def sample_dfxp_for_legacy_writer_output():
    </p>
   </div>
  </body>
-</tt>"""
+</tt>
+"""
 
 
 @pytest.fixture(scope="session")
@@ -1358,7 +1373,8 @@ def sample_dfxp_empty_cue_output():
      </p>
    </div>
  </body>
-</tt>"""
+</tt>
+"""
 
 
 @pytest.fixture(scope="session")
@@ -1432,7 +1448,8 @@ def sample_dfxp_with_ampersand_character():
    </p>
   </div>
  </body>
-</tt>"""
+</tt>
+"""
 
 
 @pytest.fixture(scope="session")

